xmrstak/backend/amd/amd_gpu/opencl/cryptonight.cl

```diff
@@ -853,6 +865,13 @@ __kernel void JOIN(cn1,ALGO) (__global uint4 *Scratchpad, __global ulong *states
 			ulong2 chunk2 = as_ulong2(SCRATCHPAD_CHUNK(2));
 			result_mul ^= chunk2;
 			ulong2 chunk3 = as_ulong2(SCRATCHPAD_CHUNK(3));
+#if(ALGO == cryptonight_v8_reversewaltz)
+			{
+				ulong2 chunk_tmp = chunk3;
+				chunk3 = chunk1;
+				chunk1 = chunk_tmp;
+			}
+#endif
```

Comment:
... you can also replace the chunk1 and chunk3 declarations here with the macro and delete this swap entirely.

Reply:
Makes sense. Thanks.
Done.